### DISCUSSION TITLE
How can I display authors?

### DISCUSSION THREAD

**Discussion by sotanakamura** [dotnet/docfx#8762] [Q&A] [answered]
Can I display authors manually or GitHub contributors automatically using metadata?

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
DocFX v3 apparently looks up contributors from GitHub ([user cache design document](https://github.com/dotnet/docfx/blob/5ff87590bc43b3a60c578ecae72c386360d174a8/docs/designs/github-user-cache.md), [graph API queries](https://github.com/dotnet/docfx/blob/c8cb790043ae5b93173f3e28dafc28bf7f305d48/src/docfx/lib/github/GitHubQueries.cs)) but AFAIK DocFX v2 does not have that feature.

If the templates in DocFX v2 were able to read a list of contributors from (per-page) metadata and display it, I imagine the metadata name and format would be documented at <https://dotnet.github.io/docfx/docs/template.html>.

If you make a custom template, I think you can use any metadata there all right.  Then you could also set up a program that examines the commit history and generates a metadata file for DocFX to read via [fileMetadataFiles](https://dotnet.github.io/docfx/reference/docfx-json-reference.html#123-separated-metadata-files-for-global-metadata-and-file-metadata), if permitted by the privacy laws in your jurisdiction.

**Reply by sotanakamura** [reply]
Thanks! I got it.
